pkg/query/storeset.go

```diff
@@ -418,11 +437,18 @@ func (s *StoreSet) getHealthyStores(ctx context.Context, stores map[string]*stor
 					level.Warn(s.logger).Log("msg", "update of store node failed", "err", errors.Wrap(err, "dialing connection"), "address", addr)
 					return
 				}
-				st = &storeRef{StoreClient: storepb.NewStoreClient(conn), cc: conn, addr: addr, logger: s.logger}
+
+				var rule storepb.RulesClient
+
+				if _, ok := ruleAddrSet[addr]; ok {
+					rule = storepb.NewRulesClient(conn)
+				}
+
+				st = &storeRef{StoreClient: storepb.NewStoreClient(conn), rule: rule, cc: conn, addr: addr, logger: s.logger}
 			}
 
 			// Check existing or new store. Is it healthy? What are current metadata?
-			labelSets, minTime, maxTime, storeType, err := spec.Metadata(ctx, st.StoreClient)
+			labelSets, minTime, maxTime, Type, err := spec.Metadata(ctx, st.StoreClient)
```

Comment:
```suggestion
			labelSets, minTime, maxTime, type, err := spec.Metadata(ctx, st.StoreClient)
```

local variables should be lower case

Reply:
oh doh, that was an old remainder of some recursive regex search/replace, thanks!